crates/oxc_ast/src/ast_impl/js.rs

```diff
@@ -665,6 +665,15 @@ impl<'a> Statement<'a> {
     }
 }
 
+impl<'a> FromIn<'a, Expression<'a>> for Statement<'a> {
+    fn from_in(expression: Expression<'a>, alloc: &'a oxc_allocator::Allocator) -> Self {
+        Statement::ExpressionStatement(Box::from_in(
```

Comment:
I would suggest using `Box::new_in` here. It's equivalent, but a little more clear.

Reply:
I explicitly used `from_in` here to keep the propagation within the `FromIn` trait implementers. This way if any major change happens to the trait it will propagate through the implementations. That's exactly why I prefer to use `IntoIn` if I were implementing an `IntoIn`. So let's say for example for some unknown reason, we would want to accept an extra parameter or maybe change the `Allocator` to `AllocationCtx`; We can migrate more easily with this pattern.

I'm open to changing it since at this point it makes no difference whatsoever. I just wanted to let you know my thought process while I was coming up with this specific implementation.